src/common/containers/builds.js

```diff
@@ -6,36 +6,52 @@ import BuildHistory from '../components/build-history';
 import { Message } from '../components/forms';
 import Spinner from '../components/spinner';
 
-import { fetchSnap } from '../actions/snap-builds';
+import { fetchBuilds, fetchSnap } from '../actions/snap-builds';
 
 import styles from './container.css';
 
 class Builds extends Component {
   fetchInterval = null
 
+  fetchData({ snapLink, repoFullName }) {
```

Comment:
If snap link is not available yet, fetch snap for given repo.
If we have the snap, fetch the builds.